connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -632,6 +636,52 @@ class SparkConnectPlanner(val session: SparkSession) {
     SerializeFromObject(udf.outputNamedExpression, mapped)
   }
 
+  private def transformFlatMapGroupsWithState(rel: proto.FlatMapGroupsWithState): LogicalPlan = {
+    val commonUdf = rel.getFunc
+    val udf = ScalaUdf(commonUdf)
+    val ds = UntypedKeyValueGroupedDataset(
+      rel.getInput,
+      rel.getGroupingExpressionsList,
+      new util.ArrayList[proto.Expression]())
+
+    val initialDs = UntypedKeyValueGroupedDataset(
+      rel.getInput,
+      rel.getInitialGroupingExpressionsList,
+      new util.ArrayList[proto.Expression]())
+
+    val hasInitialState = !rel.getInitialGroupingExpressionsList.isEmpty
+    val timeoutConf = if (rel.getTimeoutConf.isEmpty) {
+      GroupStateTimeout.NoTimeout
+    } else {
+      org.apache.spark.sql.execution.streaming.GroupStateImpl
+        .groupStateTimeoutFromString(rel.getTimeoutConf)
+    }
+    val outputMode = if (rel.getOutputMode.isEmpty) {
+      OutputMode.Update
+    } else {
+      InternalOutputModes(rel.getOutputMode)
+    }
+
+    val flatMapGroupsWithState = new FlatMapGroupsWithState(
+      udf.function.asInstanceOf[(Any, Iterator[Any], LogicalGroupState[Any]) => Iterator[Any]],
+      udf.inputDeserializer(ds.groupingAttributes),
+      ds.valueDeserializer,
+      ds.groupingAttributes,
+      ds.dataAttributes,
+      udf.outputObjAttr,
+      ds.kEncoder.asInstanceOf[ExpressionEncoder[Any]],
```

Comment:
Placeholder for state encoder, will need to investigate how to pass the state encoder when I'm back